rclpy/rclpy/node.py

```diff
@@ -632,7 +629,19 @@ def _set_parameters_atomically(
             assert len(descriptor_list) == len(parameter_list)
 
         result = None
-        if self._parameters_callback:
+        # First check if there's a read-only parameter among the ones in the list.
+        # The first time a read-only parameter is declared it has no descriptor
+        # at this point.
+        if any(
+                self.describe_parameter(param.name).read_only for
+                param in parameter_list if
+                param.name in self._descriptors
```

Comment:
nitpick: Indention looks weird here, shouldn't it only be indent 4 spaces from the column the `if` is in?

Reply:
Thanks; corrected in https://github.com/ros2/rclpy/pull/347/commits/2c9e657d3497c35037e4e80dac362bc12b532b71.